src/binary.rs

```diff
@@ -0,0 +1,198 @@
+use crate::ast::{BinSegmentOption, SegmentOptionCategory as Category, SrcSpan};
+use crate::typ::Type;
+use std::sync::Arc;
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct BinaryTypeSpecifier<T> {
+    typ: Option<BinSegmentOption<T>>,
+    signedness: Option<BinSegmentOption<T>>,
+    endianness: Option<BinSegmentOption<T>>,
+    unit: Option<BinSegmentOption<T>>,
+    size: Option<BinSegmentOption<T>>,
+}
+
+impl<T> BinaryTypeSpecifier<T> {
+    pub fn new(options: Vec<BinSegmentOption<T>>, must_have_size: bool) -> Result<Self, Error>
+    where
+        T: Clone,
+    {
+        let empty_bts = BinaryTypeSpecifier {
+            typ: None,
+            signedness: None,
+            endianness: None,
+            unit: None,
+            size: None,
+        };
+
+        let parsed_bts = options.iter().try_fold(empty_bts, |mut bts, option| {
```

Comment:
Does this mean that the options can be given in any order?

Reply:
It does, which I think is consistent with both erlang and elixir.

Reply:
TIL! Thank you